wordpress-mobile/WordPress-iOS
# Adjust PageTemplateCategories to be associated to the Blog

### Description
This is a change to associate the fetched starter page templates to be associated with the current blog. Fetching of blog layouts currently come from one of two API endpoints:
WPCOM: `/wpcom/v2/sites/\(blogID)/block-layouts` <- includes `blogID`
Self-Hosted: `/wpcom/v2/common-block-layouts`

This means that self-hosted sites, and different WPCOM sites could in the future be provided with different sets of starter page templates and categories. The previous implementation would fetch all categories when loading a site. So a user could end up with a mix-match of layouts. However, this is a bit of an edge case because the API call is made every time the picker loads, so the picker could potentially load with the wrong data then would update to the correct set of layouts when the call completes.

### To test:
📓 This is best to use a user that has multiple sites associated with their WPCom username. It's also easiest to use a fresh install to prevent any contamination of cached data from any previous testing. 
<details>
<summary><strong>To disable or enable the development version of Modal Layout Picker</strong></summary>

- Open the app from the build that allows FeatureFlags such as a PR build or a local development build
    - By default, the modal layout picker will be disabled.
    - From the site page:
        - Click on your Gravatar.
        - Click on App Settings.
        - Click on Debug.
        - Toggle "Gutenberg Modal Layout Picker" to enable or disable the picker
            <img width=300 src="https://user-images.githubusercontent.com/3384451/90816133-c8b10580-e2f9-11ea-8584-6bbfd7e523e2.gif" />
</details>

When enabled, the Layout Picker should show when creating a new page from My Site and from the Page list.

1. On site `A` navigate to the Modal Layout Picker
    - Wait for layouts to load 
1. Disconnect the device from the internet
1. On site `B` navigate to the Modal Layout Picker
    - **Expect** The page to show the Ghost cells
    - 📓 If you perform these steps on a build from `develop` you'll notice that layouts will load here.
1. On site `A` navigate to the Modal Layout Picker
    - **Expect** The page to show with the layouts that were previously fetched. 

### PR submission checklist:
